### PR TITLE
Add release notes for Chef 14.2

### PR DESCRIPTION
**Pull request by tas50**
Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,5 +1,36 @@
...
+
+## default_env Property in Execute Resource
+
+The shell_out helper has been extended with a new option `default_env` to allow disabling Chef from modifying PATH and LOCALE environmental variables as it shells out. This new option defaults to true (modify the env), preserving the previous behavior of the helper.
+
+The execute resource has also been updated with a new property `default_env` that allows utilizing this the ENV sanity functionality in shell_out. The new property defaults to false, but it can be set to true in order to ensure a sane PATH and LOCALE when shelling out. If you find that binaries cannot be found when using the execute resource, `default_env` set to true may resolve those issues.
+
+## Small Size on Disk
+
+Chef now bundles the inspec-core and train-core gems, which omit many cloud dependencies not needed within the Chef client. This change reduces the install size of a typical system by ~22% and the number of files within that installation by ~20% compared to Chef 14.1\. Enjoy the extra disk space.
```
Comment (coderanger) [inline RELEASE_NOTES.md]: An extra `\` snuck in.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,5 +1,36 @@
 This file holds "in progress" release notes for the current release under development and is intended for consumption by the Chef Documentation team. Please see <https://docs.chef.io/release_notes.html> for the official Chef release notes.
 
+# Chef Client Release Notes 14.2:
+
+## `ssh-agent` support for user keys
+
+You can now use `ssh-agent` to hold your user key when using knife. This allows storing your user key in an encrypted form as well as using `ssh -A` agent forwarding for running knife commands from remote devices.
+
+You can enable this by add `ssh_agent_signing true` to your `knife.rb` or `ssh_agent_signing = true` in your `credentials` file.
```
Comment (coderanger) [inline RELEASE_NOTES.md]: `by adding`.

File: RELEASE_NOTES.md (hunk 1 above)
Comment (tas50) [reply RELEASE_NOTES.md]: Markdown auto correct loves those for some reason

File: RELEASE_NOTES.md (hunk 2 above)
Comment (tas50) [reply RELEASE_NOTES.md]: You know you wrote this right?

Comment (coderanger) [reply RELEASE_NOTES.md]: I did mention that :) In my defense I think I wrote that whole feature in like 24 hours straight through.

**Pull request merged**